### ISSUE TITLE
Cardbook: Lookup fails when replied email address has mixed case

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/SmartTemplates#278]
reported over at #257 by @richanl:

> **One problem left:** If mail-addresses in my Cardbook are stored with capitals as Hans.Mayer@, Smart template doesn't recognize it in the Mail with %recipient(addressbook,nickname)%. Do I change it to hans.mayer@, it works on my system. Same Problem with "answer": If the senders mail-address is transmitted with capitals, ST4 doesn't recognize it. The letters must be in lower case. Can you fix this behaviour?

After I talked with the author of Cardbook this morning and [raised an issue on his Gitlab](https://gitlab.com/CardBook/CardBook/-/issues/1601), It turns out that he stores all emails in lowercase (according to standard) so this means that I can safely convert the email from any AB variable to lowercase when I look up addresses from the Cardbook database. New version will follow below.

I already made a full patch where the switch (conversion to lowercase) was optional and disabled, and I also already made a UI and full translations across all languages in the advanced preferences. I guess I can throw that work away.... it would have just been a one-liner change.

**Comment by RealRaven2000** [comment]
I have to modify the following string from the change log (and localize to the other 20 languages)

```
 {L1}New option to force lowercase emails when looking up Cardbook contacts. {L2}
```
Since it won't be an option anymore, it will be something like:

```
 {L1}Force lowercase emails when looking up Cardbook contacts. [issue 278]{L2}
```

**Comment by richanl** [comment]
> It turns out that he stores all emails in lowercase (according to standard) ....

I'm not sure about this:
cardbook v92.3:
![grafik](https://github.com/RealRaven2000/SmartTemplates/assets/88342180/604dbbea-2d0d-434f-9829-052e52055fdd)
VCARD of this:
![grafik](https://github.com/RealRaven2000/SmartTemplates/assets/88342180/8d069d88-3f50-4e22-b94f-d204574564ea)

**Comment by RealRaven2000** [comment]
Can you please test this new version below? I force lower case conversion on search in this one, it seems to work. Also Phillipe wrote in the [Gitlab comment](https://gitlab.com/CardBook/CardBook/-/issues/1601#note_1784407491):

![image](https://github.com/RealRaven2000/SmartTemplates/assets/6314674/9de392f4-bc29-477a-8053-aa3067959ccf)

new version to test:
[smartTemplate-fx-4.4pre30.zip](https://github.com/RealRaven2000/SmartTemplates/files/14374854/smartTemplate-fx-4.4pre30.zip)

I also removed the option (lcase now forced always before lookup)  and updated translations in the new change log:

![image](https://github.com/RealRaven2000/SmartTemplates/assets/6314674/5a017498-d3ff-4bb1-af07-d191d0a5e47e)

**Comment by RealRaven2000** [comment]
Implemented in 4.4 - published 28/03/2024
